src/sensai/evaluation/evaluator.py

```diff
@@ -20,10 +21,28 @@
 
 
 class MetricsDictProvider(ABC):
+    def __init__(self):
+        self.trackedExperiment: Optional[TrackedExperiment] = None
+
     @abstractmethod
     def computeMetrics(self, model, **kwargs) -> Dict[str, float]:
         pass
 
+    def setTrackedExperiment(self, trackedExperiment: TrackedExperiment):
+        self.trackedExperiment = trackedExperiment
+
+    def unsetTrackedExperiment(self):
+        self.trackedExperiment = None
+
+    def computeAndTrack(self, model, returnResult=True, **kwargs) -> Optional[Dict[str, float]]:
+        if self.trackedExperiment is None:
+            raise Exception("Cannot track results: trackedExperiment is not set.")
+        valuesDict = self.computeMetrics(model, **kwargs)
+        valuesDict["str(model)"] = str(model)
+        self.trackedExperiment.trackValues(valuesDict)
+        if returnResult:
```

Comment:
Why do we need this parameter?

Reply:
if you decide to track, you might not need to return the result. This could be called in a loop, populating some remote dashboard

Reply:
Returning a value does not cost anything: at the machine level, the address of the return value (which already exists) is merely pushed onto the stack (rather than None). Omitting it is, therefore, not useful.

Reply:
You are right, I will remove the parameter

Reply:
Done

Reply:
Done